src/server/status/metrics.js

```diff
@@ -2,70 +2,66 @@ import { get, set, isObject } from 'lodash';
 import { keysToSnakeCaseShallow } from '../../utils/case_conversion';
 import { getAllStats as cGroupStats } from './cgroup';
 
-let cGroupStatsAvailable = true;
-
-export function collectMetrics(kbnServer, server, config) {
-  server.plugins['even-better'].monitor.on('ops', event => {
-    getMetrics(event, config, server).then(data => { kbnServer.metrics = data; });
-  });
-}
+export class Metrics {
+  constructor(config, server) {
+    this.config = config;
+    this.server = server;
+    this.cGroupStatsAvailable = true;
+  }
 
-export async function getMetrics(event, config, server) {
-  const port = config.get('server.port');
-  const timestamp = new Date().toISOString();
-  const cgroup = await cGroupStatsIfAvailable();
+  async capture(event) {
```

Comment:
I like that. I updated to break out `captureEvent` and `captureCGroups`. Now `capture` simply provides some ancillary data like timestamp and merges the responses of those other functions.